src/managers/NAVManager.sol

```diff
@@ -0,0 +1,302 @@
+import {ShareClassId} from "../common/types/ShareClassId.sol";
+import {IValuation} from "../common/interfaces/IValuation.sol";
+import {ISnapshotHook} from "../common/interfaces/ISnapshotHook.sol";
+import {AccountId, withCentrifugeId} from "../common/types/AccountId.sol";
+
+import {IHub} from "../hub/interfaces/IHub.sol";
+import {IHoldings} from "../hub/interfaces/IHoldings.sol";
+import {IAccounting} from "../hub/interfaces/IAccounting.sol";
+import {IHubRegistry} from "../hub/interfaces/IHubRegistry.sol";
+
+/// @dev Assumes all assets in a pool are shared across all share classes, not segregated.
+contract NAVManager is INAVManager {
+    PoolId public immutable poolId;
+
+    IHub public immutable hub;
+    IHubRegistry public immutable hubRegistry;
+    IHoldings public immutable holdings;
+    IAccounting public immutable accounting;
+
+    INAVHook public navHook;
+    mapping(uint16 centrifugeId => uint16) public accountCounter;
+    mapping(uint16 centrifugeId => mapping(AssetId => AccountId)) public assetIdToAccountId;
+    mapping(address => bool) public manager;
+
+    constructor(PoolId poolId_, IHub hub_) {
+        poolId = poolId_;
+
+        hub = hub_;
+        hubRegistry = hub_.hubRegistry();
+        holdings = hub.holdings();
+        accounting = hub.accounting();
+    }
+
+    modifier onlyManager() {
+        require(manager[msg.sender], NotAuthorized());
+        _;
+    }
+
+    modifier onlyHubManager() {
+        require(hubRegistry.manager(poolId, msg.sender), NotAuthorized());
+        _;
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // Administration
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc INAVManager
+    function setNAVHook(INAVHook navHook_) external onlyHubManager {
+        navHook = navHook_;
+        emit SetNavHook(address(navHook_));
+    }
+
+    /// @inheritdoc INAVManager
+    function updateManager(address manager_, bool canManage) external onlyHubManager {
+        require(manager_ != address(0), EmptyAddress());
+
+        manager[manager_] = canManage;
+
+        emit UpdateManager(manager_, canManage);
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // Account creation
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc INAVManager
+    function initializeNetwork(uint16 centrifugeId) external onlyManager {
+        require(accountCounter[centrifugeId] == 0, AlreadyInitialized());
+
+        hub.createAccount(poolId, equityAccount(centrifugeId), false);
+        hub.createAccount(poolId, liabilityAccount(centrifugeId), false);
+        hub.createAccount(poolId, gainAccount(centrifugeId), false);
+        hub.createAccount(poolId, lossAccount(centrifugeId), false);
```

Comment:
Q. Are all these accounts `creditNormal`? `liabilityAccount` is `debitNormal` inside `initializeLiability()`